pandas/core/arrays/integer.py

```diff
@@ -95,6 +95,17 @@ def construct_array_type(cls) -> Type["IntegerArray"]:
         """
         return IntegerArray
 
+    def _get_common_type(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
+        # for now only handle other integer types
+        if not all(isinstance(t, _IntegerDtype) for t in dtypes):
+            return None
+        np_dtype = np.find_common_type(
```

Comment:
is there a reason you are not using pandas `pandas.core.dtypes.cast.find_common_type` ?

Reply:
Because this are all numpy integer dtypes, so there is no need to use our version that can handle our custom logic/dtypes